### PR TITLE
Filter by name now also searches on unit aliases

### PR DESCRIPTION
**Pull request by AndreMantas**

### CODE REVIEW

File: src/app/core/services/unit.service.ts [hunk 1]
```diff
@@ -5,43 +5,40 @@ import { Injectable } from '@angular/core';
...
-      unitDetail.name = window.units[i][0];
-      unitDetail.baseHp = window.units[i][12];
-      unitDetail.baseAtk = window.units[i][13];
-      unitDetail.baseRcv = window.units[i][14];
+
+      let vsUnit = false;
+      const id = i + 1;
+      const unitDetail: TempUnit = {
+        id,
+        complete: true,
```
Comment (ThePieBandit) [inline src/app/core/services/unit.service.ts:29]: Are we using complete at all at this point? is it always true?

File: src/app/core/services/unit.service.ts [hunk 2]
```diff
@@ -5,43 +5,40 @@ import { Injectable } from '@angular/core';
 
 declare const window: any;
 
+type TempUnit = Omit<UnitDetails, 'stats' | 'ability' | 'target' | 'pattern' | 'special'>;
```
Comment (ThePieBandit) [inline src/app/core/services/unit.service.ts:8]: lets give this type a proper name

File: src/app/shared/models/rumble.d.ts [hunk 3]
```diff
@@ -62,20 +62,13 @@ export interface RumbleSchema {
 }
 export interface Unit {
   ability: [Ability, Ability, Ability, Ability, Ability];
-  global?: Unit;
```
Comment (ThePieBandit) [inline src/app/shared/models/rumble.d.ts:65]: I think these need to stay....I might need to check to see if I put them in the schema.json, if not, they need to be there. This is for when units have different stats between the global and JP versions.

Comment (ThePieBandit) [reply src/app/shared/models/rumble.d.ts:65]: Don't worry about them for now, I can add them back in if necessary. We don't do anything with it yet, this was me planning for the future.

File: src/app/shared/pipes/unit-filter.pipe.ts [hunk 4]
```diff
@@ -32,10 +33,10 @@ export class UnitFilterPipe implements PipeTransform {
       filtered = filtered.filter(u => !u.isBaseForm);
     }
     if (nameFilter) {
-      // TODO: filter also by "Others" when available
```
Comment (ThePieBandit) [inline src/app/shared/pipes/unit-filter.pipe.ts:35]: See? I had a plan at some point to do this :D

**Review by ThePieBandit — APPROVED** [review]
LGTM. Have a few small comments. If you want to take a look at them, once you make changes, you can merge

File: src/app/core/services/unit.service.ts (hunk 1 above)
Comment (AndreMantas) [reply src/app/core/services/unit.service.ts:29]: yeah the "complete" concept goes away when we decided to stick with the above `if (window.units[i].incomplete) { continue; }`

File: src/app/core/services/unit.service.ts (hunk 2 above)
Comment (AndreMantas) [reply src/app/core/services/unit.service.ts:8]: i dont have a better name for it 😄 
its rly a temporary object that represents a unit that does not have some properties yet and was the best TS-way i could do it

File: src/app/shared/models/rumble.d.ts (hunk 3 above)
Comment (AndreMantas) [reply src/app/shared/models/rumble.d.ts:65]: yeah i checked on the raw json and none of the objects there had these properties.. so to to make it "match the auto generated types from json" it made sense to delete

Comment (ThePieBandit) [reply src/app/shared/models/rumble.d.ts:65]: I just checked, it looks like they're in the optc-db version of the schema. I need to set it up to auto pull and auto regen rumble.d.ts

File: src/app/core/services/unit.service.ts (hunk 1 above)
Comment (ThePieBandit) [reply src/app/core/services/unit.service.ts:29]: ok, we'll remove this in the future then.

**Pull request merged**